PROTO_tests/tests/numeric_test.py

```diff
@@ -212,6 +212,18 @@ def test_sin(self, num_type):
         pda = ak.array(na, dtype=num_type)
 
         assert np.allclose(np.sin(na), ak.sin(pda).to_ndarray(), equal_nan=True)
+
+        # Test where feature for singleton and array cases
+        truth_np = np.arange(len(na)) % 2 == 0
+        truth_ak = ak.array(truth_np)
+        assert np.allclose(np.sin(na, where=True), ak.sin(pda, where=True).to_ndarray(), equal_nan=True)
+        assert np.allclose(na, ak.sin(pda, where=False).to_ndarray(), equal_nan=True)
+        assert np.allclose(
+            [np.sin(na[i]) if truth_np[i] else na[i] for i in range(len(na))],
+            ak.sin(pda, where=truth_ak).to_list(),
+            equal_nan=True,
+        )
```

Comment:
Since the only thing changing here is the trig func, i think let's pull all the trig where versions out into their own test and we can just look over `func`. I have examples of this that i can point you to when you get back

Reply:
Updated